pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -504,7 +514,10 @@ func checkAndPublishDhcpLeases(ctx *zedrouterContext) {
 				log.Functionf("found %t IP %s for %s",
 					assigned, leasedIP.String(), ulStatus.Mac)
 			}
-			assignedIP := net.ParseIP(ulStatus.AllocatedIPAddr)
+			assignedIP := net.IP{}
+			if len(ulStatus.AllocatedIPAddr) != 0 {
+				assignedIP = net.ParseIP(ulStatus.AllocatedIPAddr[0])
```

Comment:
Seems to be lots of assumptions that [0] is the IPv4 address. That seems a bit brittle.

Reply:
removed this assumption by splitting addresses into ip version specific lists